addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/cache/ExpiringCacheAsync.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.cache;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Complementary class to {@see org.eclipse.smarthome.core.cache.ExpiringCache}, implementing an async variant
+ * of an expiring cache. Returns the cached value immediately to the callback if not expired yet, otherwise issue
+ * a fetch and notify callback implementors asynchronously.
+ *
+ * @author David Graeff
+ *
+ * @param <V> the type of the value
+ */
+public class ExpiringCacheAsync<V> {
+    private final long expiry;
+    private final ExpiringCacheUpdate cacheUpdater;
+    private long expiresAt = 0;
+    private V value;
+    private List<ExpiringCacheValue<V>> waitingCacheCallbacks = new LinkedList<>();
+
+    /**
+     * Implement this interface to be able to request the value of {@see ExpiringCacheAsync}.
+     * The value will be returned immediately if not expired or first fetched and you will be notified
+     * as soon as there is an updated value.
+     *
+     */
+    public static interface ExpiringCacheValue<V> {
+        void cacheValue(V value);
+    }
+
+    /**
+     * Implement the requestCacheUpdate method which will be called when the cache
+     * needs an updated value. Call {@see setValue} to update the cached value.
+     */
+    public static interface ExpiringCacheUpdate {
+        void requestCacheUpdate();
+    }
+
+    /**
+     * Create a new instance.
+     *
+     * @param expiry the duration in milliseconds for how long the value stays valid
+     * @param cacheUpdater The cache will use this callback if a new value is needed.
+     */
+    public ExpiringCacheAsync(long expiry, ExpiringCacheUpdate cacheUpdater) {
```

Comment:
Null check for `cacheUpdater` to prevent NPE in [refreshValue](https://github.com/openhab/openhab2-addons/pull/2380/files#diff-e4d04165f1465214d41ca00e2de0a61cR105)?